### PR TITLE
Perishable pet food

### LINKED ISSUE #31076: Cat and dog food lasting forever doesn't make sense
# Describe the bug
Cat and dog food are relativity easy to make (cooking 4) they do not perish, despite not requiring any tools such as a canning pot to make, how exacting does combining cooked meat with water make something not spoil?

# Steps To Reproduce

1. Get some raw meat and water
2. Cook some cat food 
3.  Now you have food that never spoils, is healthy to eat, and only has a minor downside of enjoyablility.

# Expected behavior
For cat/dog food to spoil or have a believable reason for it to have a indefinite shelf life.

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1809
- Game Version: 0.D-3778-gcec1ebf [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Folding Parts pack [deoxymod],
    Classes and Scenarios Mod [more_classes_scenarios],
    Alternative Map Key [alt_map_key],
    Mythical Martial Arts [MMA]
]

### PR DESCRIPTION
**Pull request by bandti45**
**Summary**
```SUMMARY: Content "Making dog and cat food perishable"```

**Purpose of change**
Fixes #31076 Makes cat and dog food perishable and ways to store it

**Describe the solution**
Add perishableness to cat and dog food (of one day since its made of cooked meat and water). Also, add canned versions with recipes to allow long term storage of said food. Change the spawns of cat and dog food to the canned version.

**Describe alternatives you've considered**
Just make it perishable for a week so it's not going bad before you have a chance to use it.

**Additional context**
I'm not sure if I did the item spawning right so feel free to give me tips

### CODE REVIEW

**Comment by NickGrease** [comment]
Under Purpose of Change, remove the back ticks ( ` ) to get "Fixes #" to work correctly.

**Comment by anothersimulacrum** [comment]
You should remove the `[` & `]` between the `""` in the summary line.

**Comment by NickGrease** [comment]
I don't know how to 'read' json recipes, but to me it appears you are missing `"result": "dogfood"` - there is `"result": "catfood"` plus canned variants, but only canned variants of dogfood.

**Comment by bandti45** [comment]
I'm not missing anything in those, the thing is since some of the values between cat and dog food are the same when I put canned dog food recipes the changes were only added where it was necessary, not how I changed the code when editing

File: data/json/items/comestibles/other.json [hunk 1]
```diff
@@ -671,4 +695,4 @@
     "flags": "TRADER_AVOID",
     "volume": 1
   }
-]
+]
```
Comment (anothersimulacrum) [inline data/json/items/comestibles/other.json]:
```suggestion
]

```

**Comment by FulcrumA** [comment]
Non/slow perishable dry kibble to be found over the world would be nice too, though.

**Comment by bandti45** [comment]
> Non/slow perishable dry kibble to be found over the world would be nice too, though.

I'm afraid that would require me to adjust all the spawning table with ratios for both which I have no idea how to balance properly and decide better numeric values for both which is something I would rather not do in this PR, sorry

**Comment by Night-Pryanik** [comment]
Please also update dogfood spawning in the island prison.

**Comment by kevingranade** [comment]
Some formatting errors in the json.

**Comment by bandti45** [comment]
I know I need to A. find the rest of the spawns that currently make uncanned cat and dog food and B. try linting the whole file again. Just been busy but I will get to it

**Comment by kevingranade** [comment]
No problem, no pressure, just a FYI :+1:

**Pull request merged**